### ISSUE TITLE
fontMetric.charMap doesn't cache

### ISSUE THREAD

**Issue by rksm** [LivelyKernel/lively.morphic#34]
@tylervernonsmith 

Currently `fontMetric.charMap` tries to cache computed boundaries by putting a "style" into the charMap object. However, "style" is an ordinary JS object. When you try to assign a JS object as the property of another like `charMap[style] = x`, `style` gets converted into a string, by default `"[object Object]"` and that is used as the key.

This means that all styles you try to cache are converted into `"[object Object]"`, i.e. essentially nothing is cached.

You have two options: You can either use `Map` but you have to make sure that the "style keys" resolve to identical objects because:

``` js
var cache = new Map();
var key = {foo: 23};
cache.set(key, 3);

cache.get(key) // => 3
cache.get({foo: 23}) // => undefined

// object identity is used for lookup!
```

Alternatively, what I would prefer, convert style into a string:

``` js
var key = `${style.fontFamily}-${style.fontSize}-...`
```

and use that as the lookup object for char map.

**Comment by rksm** [comment]
Another alternative is to use a nested structure `charMap[fontFamily][fontSize][boldness][decoration]...` for caching. That's probably nicer.

**Comment by tylervernonsmith** [comment]
Thanks Robert, fixed in 26c5facbaf39a39259520214f1d4bd43a58fc838.
